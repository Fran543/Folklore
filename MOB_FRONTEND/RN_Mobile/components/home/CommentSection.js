import { View, StyleSheet} from "react-native";
import CommentTextInput from '../ui/CommentTextInput'
import CommentsHolder from "../home/CommentsHolder";
import CommentSubmitButton from "../ui/CommentSubmitButton";
import {useState, useEffect} from 'react'

export default function CommentSection(props) {
    const [commentText, setCommentText] = useState()

    function commentTextChangeHandler(textInputValue){
        setCommentText(textInputValue)
        console.log(commentText)
    }

    return (
        <View style={styles.container}>
            <CommentTextInput onChangeText={commentTextChangeHandler} commentText={commentText}/>
            <CommentSubmitButton commentText={commentText} idStory={props.idStory}/>
            <CommentsHolder idStory={props.idStory}/>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        paddingHorizontal: 20,
        paddingBottom: 20
    },
}); 