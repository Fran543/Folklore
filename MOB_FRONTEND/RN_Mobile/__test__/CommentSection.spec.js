import React from 'react';
import renderer from 'react-test-renderer';

import CommentSection from '../components/home/CommentSection';

describe('<CommentSection />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<CommentSection idStory="1"/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has 3 child', () => {
        const tree = renderer.create(<CommentSection idStory="1"/>).toJSON();
        expect(tree.children.length).toBe(3);
    });
});
