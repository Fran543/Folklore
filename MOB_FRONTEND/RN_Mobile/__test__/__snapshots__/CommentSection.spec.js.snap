// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentSection /> renders correctly 1`] = `
<View
  style={
    Object {
      "paddingBottom": 20,
      "paddingHorizontal": 20,
    }
  }
>
  <TextInput
    keyboardType="default"
    maxLength={500}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Enter your comment here..."
    style={
      Object {
        "borderColor": "#808080",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "white",
        "height": 100,
        "padding": 12,
      }
    }
    textAlign="left"
  />
  <View
    style={
      Object {
        "borderRadius": 28,
        "margin": 4,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#72063C",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "letterSpacing": 3,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Submit
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "paddingTop": 30,
      }
    }
  >
    <RCTScrollView
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
